core/src/main/java/org/apache/calcite/sql/fun/SqlStdOperatorTable.java

```diff
@@ -1872,10 +1901,26 @@ public class SqlStdOperatorTable extends ReflectiveSqlOperatorTable {
       // Return type is same as divisor (2nd operand)
       // SQL2003 Part2 Section 6.27, Syntax Rules 9
       SqlBasicFunction.create(SqlKind.MOD,
+          // A rather unfortunate name for this return type strategy
           ReturnTypes.NULLABLE_MOD,
           OperandTypes.EXACT_NUMERIC_EXACT_NUMERIC)
           .withFunctionType(SqlFunctionCategory.NUMERIC);
 
+  /**
+   * Variant of arithmetic remainder function {@code MOD} which returns NULL when
+   * the denominator is 0.
+   */
+  public static final SqlFunction MOD_0_NULL =
+      // Return type is same as divisor (2nd operand)
+      // SQL2003 Part2 Section 6.27, Syntax Rules 9
+      // Unfortunately there cannot exist two functions in the standard operator
+      // table with the same exact name, so we need to use a different name,
+      // although in SQL this would be shown just as MOD
+      SqlBasicFunction.create("NULLABLE_MOD", SqlKind.MOD_0_NULL,
```

Comment:
There is also `PERCENT_REMAINDER` with name `%` why did we pick "NULLABLE_MOD" as a name?